### ISSUE TITLE
Tests build failure due to removal of std::random_shuffle() in C++17

### ISSUE THREAD

**Issue by sssilver** [gabime/spdlog#1041]
When building with clang x86_64-apple-darwin18.5.0 on latest macOS, I get the following error:

```
[9/41] Compiling C++ object 'subprojects/spdlog-1.3.1/tests/7c1051b@@tests_exe@exe/main.cpp.o'.
FAILED: subprojects/spdlog-1.3.1/tests/7c1051b@@tests_exe@exe/main.cpp.o
c++ -Isubprojects/spdlog-1.3.1/tests/7c1051b@@tests_exe@exe -Isubprojects/spdlog-1.3.1/tests -I../subprojects/spdlog-1.3.1/tests -I../subprojects/spdlog-1.3.1/include -Xclang -fcolor-diagnostics -pipe -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++17 -g  -MD -MQ 'subprojects/spdlog-1.3.1/tests/7c1051b@@tests_exe@exe/main.cpp.o' -MF 'subprojects/spdlog-1.3.1/tests/7c1051b@@tests_exe@exe/main.cpp.o.d' -o 'subprojects/spdlog-1.3.1/tests/7c1051b@@tests_exe@exe/main.cpp.o' -c ../subprojects/spdlog-1.3.1/tests/main.cpp
In file included from ../subprojects/spdlog-1.3.1/tests/main.cpp:2:
../subprojects/spdlog-1.3.1/tests/catch.hpp:5796:26: error: no member named 'random_shuffle' in namespace 'std'
                    std::random_shuffle( matchingTestCases.begin(), matchingTestCases.end(), rng );
                    ~~~~~^
1 error generated.
```

`std::random_shuffle()` does indeed seem to be deprecated in C++14 and removed in C++17.

What's the workaround for this?

**Comment by gabime** [comment]
spdlog uses pretty old version of catch lib. I wonder if upgrading to latest version would solve this.

**Comment by gabime** [comment]
Fixed in e39959a132af12cf8fad0e07e23f40e8ccf55312
